source/develop/release-management/features/storage/incremental-backup.html.md

```diff
@@ -12,173 +12,421 @@ feature_status: Design
+snapshot, merging the top qcow2 layer into the base layer will convert
+the disk to raw, and disable incremental backup. The user can create a
+new snapshot including this disk to enable back incremental backup.
+
+### Disk format
+
+Here is a table showing how enabling incremental backup affects disk
+format.
+
+```
+storage     provisioning        incremental     format
+====================================================================
+block       thin                enabled         qcow2
+block       preallocated        enabled         qcow2 (preallocated)
+file        thin                enabled         qcow2
+file        preallocated        enabled         qcow2 (preallocated)
+--------------------------------------------------------------------
+block       thin                disabled        qcow2
+block       preallocated        disabled        raw (preallocated)
+file        thin                disabled        raw (sparse)
+file        preallocated        disabled        raw (preallocated)
+--------------------------------------------------------------------
+network     -                   disabled        raw
+lun         -                   disabled        raw
+```
+
+### Incremental backup flow
 
+1. Backup application finds VM disks enabled for incremental backup
+   using oVirt API.
 
-## Incremental backup flow
-Backup flow is the same from the point of view of the user (backup
-application), but will be implemented differently under the hood.
+1. Backup application starts backup using oVirt API, specifying a VM id,
+   optional previous backup id, and list of disks to backup.
+   If previous backup uuid isn't specified, all data in the disks at the
+   point of the backup will be included in the backup.
 
-1. Backup application starts backup using oVirt API, specifying the last
-backup. The backup will include blocks that changed since that backup.
-2. System prepares VM for backup for running VM, or the VM disks for
-backup for a stopped VM.
-3. Backup application starts image transfer for download
-4. Backup application obtains changed blocks list from ovirt-imageio
-5. Backup application downloads changed blocks from ovirt-imageio
-6. Backup application finalize image transfer
-7. Backup application finish backup using oVirt API
+1. System prepares VM for backup. The VM may be running during the
+   backup.
 
+1. Backup application waits until backup is ready using oVirt SDK.
 
-## Create VM with incremental backup support
-To allow incremental backup the user must enable the feature per VM.
+1. When backup is ready backup application creates image transfer for
+   every disk included in the backup.
 
-### Open issues
-Do we need to support per disk incremental backup?
+1. Backup application obtains changed blocks list from ovirt-imageio for
+   every image transfer.
 
+1. Backup application downloads changed blocks in raw format from
+   ovirt-imageio and store them in the backup media.
```

Comment:
If no change list is available, backup application will fallback to full backup.